utils/src/main/kotlin/storage/HttpFileStorage.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.utils.storage
+
+import com.here.ort.utils.OkHttpClientHelper
+import com.here.ort.utils.getUserOrtDirectory
+
+import java.io.IOException
+import java.io.InputStream
+import java.net.HttpURLConnection
+import java.util.concurrent.TimeUnit
+
+import okhttp3.CacheControl
+import okhttp3.Headers.Companion.toHeaders
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+
+/**
+ * A [FileStorage] that stores files on an HTTP server.
+ */
+class HttpFileStorage(
+    /**
+     * The URL to store files at.
+     */
+    private val url: String,
+
+    /**
+     * Custom headers that are added to all HTTP requests.
+     */
+    private val headers: Map<String, String> = emptyMap(),
+
+    /**
+     * A path inside the [ORT user directory][getUserOrtDirectory] used for caching HTTP responses. Defaults to
+     * "cache/http".
+     */
+    private val cachePath: String = "cache/http",
+
+    /**
+     * The max age of an HTTP cache entry in seconds. Defaults to 0.
```

Comment:
I'm not sure if I understand the default value correctly? would it make sense to say: "Defaults to 0 which disables caching" ?

Reply:
Done.